### PR TITLE
feat(contacts): linked contacts with the database and navigation flow refinement

### PR DESCRIPTION
**Pull request by gaspardthrl**
# What I did
- I linked the contact part with the database
- I implemented new animations and gestures and refined existing ones
- I modified the navigation flow
- I changed the nomenclature of files to match with this new navigation workflow

# How I did it
- I used existing functions to retrieve current user from the database
- I implemented new functions and hooks to ensure a smooth flow with no overhead when fetching new user (either friends of yours or friends of your friends)
- I used the Animation elements from react-native

# How to verify it
### I would advise you to use the following credentials : 
- gasthoral@gmail.com
- Abcdefg123

### I will also define more explicitly the two following terms:
- Long press => Press a node and don't release your finger until you feel a vibration (if you have them enabled on your phone) or simply wait for either a toast message (pop up at the top of the screen) to appear or for an animation to launch
- Double-click => Double tap very quickly

1. Check that when you first open the app (given that you are already logged in) you see the graph view first.
2. Check that long pressing the center node results in an error toast message
3. Check that long pressing on the friend's node named "Gustave Charles" launches an animation and that a new node appear (actually there is a 25% chance that it won't happen as we use a random mechanism to make the graph generation more "natural" so you might need to long press the node again to come back to beginning of point 3)
4. When you see a new node appear test that long pressing it raises an error toast message.
5. Check that double clicking on a node displays it modal from the top and that clicking outside the modal dismisses it by it sliding upward.

# Demo video

https://github.com/uniconnect-epfl/uniconnect/assets/81919237/8c43570f-272c-408f-97f5-72513e240e69

# Pre-merge checklist
The changes I have introduced:
- [x] work correctly
- [x] do not break other functionalities
- [x] work correctly on Android
- [x] are fully tested

### CODE REVIEW

File: components/TabBar/TabBar.tsx [hunk 1]
```diff
@@ -20,47 +30,47 @@ export const TabBar = ({ state, descriptors, navigation }: BottomTabBarProps) =>
           options.tabBarLabel !== undefined
             ? options.tabBarLabel.toString()
             : options.title !== undefined
-              ? options.title
-              : route.name
+            ? options.title
+            : route.name
```
Comment (AlbertoCentonze) [inline components/TabBar/TabBar.tsx:34]: Ternary operators can be made a lot more readable using modern javascript:
```
options.tabBarLabel?.toString() ?? options.title ?? route.name
```

should do the same!

File: screens/Contacts/NetworkScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        setGraph(graph)
+      })
+    }
+  }, [contacts])
+
+  useEffect(() => {
+    let tempUserId = getAuth().currentUser?.uid
+    if (tempUserId === undefined) {
+      tempUserId = "fwLsAGVKvqZl3gb0S28bDtN6Yvd2"
+    }
```
Comment (AlbertoCentonze) [inline screens/Contacts/NetworkScreen.tsx:67]: Same thing here: 
```
let tempUserId = getAuth().currentUser?.uid ?? "fwLsAGVKvqZl3gb0S28bDtN6Yvd2"
```

**Review by AlbertoCentonze — CHANGES_REQUESTED** [review]
Impeccable job as usual, just a couple of instances where cognitive complexity can be reduced in the code!

File: components/TabBar/TabBar.tsx (hunk 1 above)
Comment (gaspardthrl) [reply components/TabBar/TabBar.tsx:34]: I will address this thanks

File: components/Graph/NodeModal/NodeModal.tsx [hunk 3]
```diff
@@ -16,41 +18,93 @@ const NodeModal: React.FC<{
...
+    if (visible) {
+      setModalVisible(true)
+      slideInModal()
+    } else {
+      setTimeout(() => {
+        setModalVisible(false)
+      }, 1000)
+    }
+  }, [visible])
+
```
Comment (Fyuryy) [inline components/Graph/NodeModal/NodeModal.tsx:56]: Don't forget to solve this issues

File: components/TabBar/TabBar.tsx [hunk 4]
```diff
@@ -7,60 +7,65 @@ import { peach } from "../../assets/colors/colors"
...
       {state.routes.map((route, index) => {
         const { options } = descriptors[route.key]
         const label =
-          options.tabBarLabel !== undefined
-            ? options.tabBarLabel.toString()
-            : options.title !== undefined
-              ? options.title
-              : route.name
-
+          options.tabBarLabel?.toString() ?? options.title ?? route.name
```
Comment (Fyuryy) [inline components/TabBar/TabBar.tsx:30]: Comments wouldn't hurt here!

File: screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx [hunk 5]
```diff
@@ -143,11 +144,15 @@ const AuthenticationScreen: React.FC = () => {
...
+              nextScreen="ExploreTabs"
+              buttonText="Confirm"
+              authenticate={submitForm}
+            />
           </View>
-        }
+        )}
       </View>
     </TouchableWithoutFeedback>
   )
```
Comment (Fyuryy) [inline screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx:158]: Let's try to have the same code formatter so we don't get these repeated parts of code in the PRs.

**Review by Fyuryy — COMMENTED** [review]
Good job Gaspard! As I've mentioned in one of your older PRs, you should keep it lighter. 48 files is huge. You could have separated the graph logic and the database interactions. Try to do it next time!

File: components/TabBar/TabBar.tsx (hunk 4 above)
Comment (gaspardthrl) [reply components/TabBar/TabBar.tsx:30]: It is a change suggested by Alberto. It is explained in the discussions here it has to do with how JS syntax works now

File: screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx (hunk 5 above)
Comment (gaspardthrl) [reply screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx:158]: I will check into it but this is weird that we don't have the same

File: components/Graph/NodeModal/NodeModal.tsx (hunk 3 above)
Comment (gaspardthrl) [reply components/Graph/NodeModal/NodeModal.tsx:56]: done

**Review by AlbertoCentonze — APPROVED** [review]

**Pull request merged**